src/Common/Integrations/Plugin_Merge_Provider_Abstract.php

```diff
@@ -103,8 +103,10 @@ public function get_plugin_real_path(): string {
 			}
 		}
 
-		// Memoize the result.
-		$plugins_path[ $text_domain ] = $plugin_path;
+		// Memoize the result if we found it.
+		if ( $plugin_path ) {
+			$plugins_path[ $text_domain ] = $plugin_path;
+		}
```

Comment:
A slightly different approach - 

```suggestion
	// Return empty string if no matching plugin is found.
	if ( empty( $plugin_path ) ) {
		return '';
	}

	// Memoize the result if we found it.
	$plugins_path[ $text_domain ] = $plugin_path;
	return $plugin_path;
```